commit 4224928879ee
Author: Michael Kaply <mozilla@kaply.com>
Date:   Wed Mar 14 13:30:53 2018 -0500

    Bug 1372107 - Allow browser to be started despite Autoconfig errors. r=florian, a=ritu
    
    MozReview-Commit-ID: of8cdUMedC
    
    --HG--
    extra : rebase_source : d58039b1f2e268c7a0af77e9a0680f759f203d99
    extra : source : dad020b0bb45903972c2856cc3a2968f39a27635
---
 extensions/pref/autoconfig/src/nsReadConfig.cpp | 27 +++++++++++++------------
 1 file changed, 14 insertions(+), 13 deletions(-)

diff --git extensions/pref/autoconfig/src/nsReadConfig.cpp extensions/pref/autoconfig/src/nsReadConfig.cpp
index a87b5b2c1bd4..8939fdc268eb 100644
--- extensions/pref/autoconfig/src/nsReadConfig.cpp
+++ extensions/pref/autoconfig/src/nsReadConfig.cpp
@@ -36,35 +36,35 @@ extern nsresult CentralizedAdminPrefManagerInit();
 extern nsresult CentralizedAdminPrefManagerFinish();
 
 
-static void DisplayError(void)
+static nsresult DisplayError(void)
 {
     nsresult rv;
 
     nsCOMPtr<nsIPromptService> promptService = do_GetService("@mozilla.org/embedcomp/prompt-service;1");
     if (!promptService)
-        return;
+        return NS_ERROR_FAILURE;
 
     nsCOMPtr<nsIStringBundleService> bundleService = do_GetService(NS_STRINGBUNDLE_CONTRACTID);
     if (!bundleService)
-        return;
+        return NS_ERROR_FAILURE;
 
     nsCOMPtr<nsIStringBundle> bundle;
     bundleService->CreateBundle("chrome://autoconfig/locale/autoconfig.properties",
                                 getter_AddRefs(bundle));
     if (!bundle)
-        return;
+        return NS_ERROR_FAILURE;
 
     nsXPIDLString title;
     rv = bundle->GetStringFromName("readConfigTitle", getter_Copies(title));
     if (NS_FAILED(rv))
-        return;
+        return rv;
 
     nsXPIDLString err;
     rv = bundle->GetStringFromName("readConfigMsg", getter_Copies(err));
     if (NS_FAILED(rv))
-        return;
+        return rv;
 
-    promptService->Alert(nullptr, title.get(), err.get());
+    return promptService->Alert(nullptr, title.get(), err.get());
 }
 
 // nsISupports Implementation
@@ -101,12 +101,13 @@ NS_IMETHODIMP nsReadConfig::Observe(nsISupports *aSubj
     if (!nsCRT::strcmp(aTopic, NS_PREFSERVICE_READ_TOPIC_ID)) {
         rv = readConfigFile();
         if (NS_FAILED(rv)) {
-            DisplayError();
-
-            nsCOMPtr<nsIAppStartup> appStartup =
-                do_GetService(NS_APPSTARTUP_CONTRACTID);
-            if (appStartup)
-                appStartup->Quit(nsIAppStartup::eAttemptQuit);
+            rv = DisplayError();
+            if (NS_FAILED(rv)) {
+                nsCOMPtr<nsIAppStartup> appStartup =
+                    do_GetService(NS_APPSTARTUP_CONTRACTID);
+                if (appStartup)
+                    appStartup->Quit(nsIAppStartup::eAttemptQuit);
+            }
         }
     }
     return rv;
